tests/integration/targets/ec2_eip/tasks/update_reverse_dns_record.yml

```diff
@@ -0,0 +1,117 @@
+- name: Test EIP allocation and reverse DNS record operations
+  block:
+    # ------------------------------------------------------------------------------------------
+    # Allocate EIP with reverse DNS record - check mode
+    # ------------------------------------------------------------------------------------------
+    - name: Allocate a new EIP and modify it's reverse DNS record - check_mode
+      amazon.aws.ec2_eip:
+        state: present
+        domain_name: "{{ test_domain }}"
+        tags: "{{ eip_test_tags }}"
+      register: eip
+      check_mode: true
+
+    - name: Assert that task result was as expected
+      ansible.builtin.assert:
+        that:
+          - eip is changed
+
+    - name: Ensure no new EIP was created
+      ansible.builtin.include_tasks: tasks/common.yml
+      vars:
+        has_no_new_eip: true
+
+    # ------------------------------------------------------------------------------------------
+    # Allocate EIP with reverse DNS record
+    # ------------------------------------------------------------------------------------------
+    - name: Allocate a new EIP and modify it's reverse DNS record
+      amazon.aws.ec2_eip:
+        state: present
+        domain_name: "{{ test_domain }}"
+        tags: "{{ eip_test_tags }}"
+      register: eip
+
+    - name: Add EIP IP address an A record
```

Comment:
Do we really need this step to test the newly added parameter? While it is beneficial to have an additional test for full functionality, I believe that if `ec2_eip` supports adding or modifying the EIP with a domain name and runs correctly in our CI account, we might not need to do the Route53 testing. What do you think? cc @alinabuzachis @abikouo